vm.go

```diff
@@ -351,6 +353,51 @@ func (vm *VirtualMachineSpec) ExecuteCommand(command string) error {
 	return nil
 }
 
+// RunCommand executes a command on the Virtual Machine
+func (vm *VirtualMachineSpec) RunCommand(command string) (output string, err error) {
+	ctx, cancel := context.WithTimeout(context.Background(), 60*time.Minute)
+	defer cancel()
+	system, err := hcs.OpenComputeSystem(ctx, vm.ID)
+	if err != nil {
+		return "", err
+	}
+	defer system.Close()
+
+	commandArray := strings.Split(command, " ")
+	params := &hcsschema.ProcessParameters{
```

Comment:
>params [](http://example.com/codeflow?start=1&length=6)

what's the user context to run this process?  always root?  Or caller should specify it?

Reply:
in this case it is root. we can make it more generic when we sync up with upstream. they have added functionality to run cmds on VM from nodectl, but that has implementation only for VMMS. i can make the changes for HCS, and we can contribute to their codebase...

Reply:
let me know if we this should be done now. can do ...

Reply:
if upstream already defined the interface for the same functionality, we should just use it.